### PR TITLE
Changed `Psbt` to `SpendTransaction`.

### LINKED ISSUE #79: ui: import spend_transaction must check size of imported PSBT
- [ ] change https://github.com/edouardparis/revault-gui/blob/master/src/app/state/manager.rs#L441 for SpendTransaction
- [ ] update parse_psbt fro SpendTransaction::from_psbt_str 
- [ ] introduce new test (check tests/app_manager.rs)

### PR DESCRIPTION
**Pull request by Zshan0**
`ManagerImportSendTransactionState` currently stores the psbt directly in `psbt_imported`. This has been updated to `SpendTransaction`. Following this change is a chain of minor changes under `impl State` for the same.

The implemented test only checks if the parsing for a `psbt` string happens through `SpendTransaction::from_psbt_str`. This PR resolves #79

### CODE REVIEW

File: tests/app_manager.rs [hunk 1]
```diff
@@ -145,3 +145,35 @@ async fn test_manager_create_spend() {
         .await;
     assert_eq!(sandbox.state().input_amount(), 119965368);
 }
+
+#[tokio::test]
+async fn test_manager_import_spend() {
+    let psbt_string :String = "cHNidP8BAIkCAAAAAUeuD/NEqc88sk3DoBrKoVKjXbN2xW8Jr/4GO5q87JqJAQAAAAD9////AriGJgcAAAAAIgAgSOjPZes2prPdrcgiv+IG1sjXyTCc4KDr9+C9F+xk6LwwdQAAAAAAACIAIAjkMa8elv7dHUmYpDATWBtmMmpv9yyKFawMunvGQ1AMAAAAAAABASsADicHAAAAACIAIHXyaRd0yBZ3gxhGsCgiAOKIssWXELWPdDGD1JJVB9vFAQMEAQAAAAEFR1IhAlgt7b9E9GVk5djNsGdTbWDr40zR0YAc/1G7+desKJtDIQNHBN7LVbWqiP/R710GNmJIwTFOGWVRE2/xTquLukpJDlKuIgYCWC3tv0T0ZWTl2M2wZ1NtYOvjTNHRgBz/Ubv516wom0MI1n1/6QAAAAAiBgNHBN7LVbWqiP/R710GNmJIwTFOGWVRE2/xTquLukpJDghyqV8iAAAAAAAiAgICkzqxA36tCqSnhYxtSdZwXh+zvF9msAkYr3ufAOzVJgglHWAJAAAAACICAlgt7b9E9GVk5djNsGdTbWDr40zR0YAc/1G7+desKJtDCNZ9f+kAAAAAIgIDRwTey1W1qoj/0e9dBjZiSMExThllURNv8U6ri7pKSQ4IcqlfIgAAAAAAIgICUHL04HZXilyJ1B118e1Smr+S8c1qtja46Le7DzMCaUMI+93szQAAAAAA".to_string();
```
Comment (edouardparis) [inline tests/app_manager.rs:151]: Can you please run `rustfmt` on the file changed

**Review by edouardparis — APPROVED** [review]
looks good !
Just run rustfmt and some nit

I approved the change but the tests are not good

File: tests/app_manager.rs [hunk 2]
```diff
@@ -145,3 +145,35 @@ async fn test_manager_create_spend() {
...
+        Role::Stakeholder,
+        Menu::DelegateFunds,
+        Box::new(|| Box::pin(no_hardware_wallet())),
+    );
+    let sandbox = sandbox
+        .update(
+            &ctx,
+            Message::SpendTx(SpendTxMessage::PsbtEdited(psbt_string)),
+        )
+        .await;
```
Comment (edouardparis) [inline tests/app_manager.rs:174]: you should check that the state has received the good psbt, you can retrieve the state with `sandbox.state()` see tests/app_stakeholder for inspiration.

File: tests/app_manager.rs [hunk 3]
```diff
@@ -145,3 +145,35 @@ async fn test_manager_create_spend() {
...
+
+    let _sandbox = sandbox
```
Comment (edouardparis) [inline tests/app_manager.rs:176]: In order to emulate the `set_spend_transaction` , the daemon request->response must be mocked

File: tests/app_manager.rs [hunk 4]
```diff
@@ -145,3 +145,35 @@ async fn test_manager_create_spend() {
         .await;
     assert_eq!(sandbox.state().input_amount(), 119965368);
 }
+
+#[tokio::test]
+async fn test_manager_import_spend() {
+    let psbt_string :String = "cHNidP8BAIkCAAAAAUeuD/NEqc88sk3DoBrKoVKjXbN2xW8Jr/4GO5q87JqJAQAAAAD9////AriGJgcAAAAAIgAgSOjPZes2prPdrcgiv+IG1sjXyTCc4KDr9+C9F+xk6LwwdQAAAAAAACIAIAjkMa8elv7dHUmYpDATWBtmMmpv9yyKFawMunvGQ1AMAAAAAAABASsADicHAAAAACIAIHXyaRd0yBZ3gxhGsCgiAOKIssWXELWPdDGD1JJVB9vFAQMEAQAAAAEFR1IhAlgt7b9E9GVk5djNsGdTbWDr40zR0YAc/1G7+desKJtDIQNHBN7LVbWqiP/R710GNmJIwTFOGWVRE2/xTquLukpJDlKuIgYCWC3tv0T0ZWTl2M2wZ1NtYOvjTNHRgBz/Ubv516wom0MI1n1/6QAAAAAiBgNHBN7LVbWqiP/R710GNmJIwTFOGWVRE2/xTquLukpJDghyqV8iAAAAAAAiAgICkzqxA36tCqSnhYxtSdZwXh+zvF9msAkYr3ufAOzVJgglHWAJAAAAACICAlgt7b9E9GVk5djNsGdTbWDr40zR0YAc/1G7+desKJtDCNZ9f+kAAAAAIgIDRwTey1W1qoj/0e9dBjZiSMExThllURNv8U6ri7pKSQ4IcqlfIgAAAAAAIgICUHL04HZXilyJ1B118e1Smr+S8c1qtja46Le7DzMCaUMI+93szQAAAAAA".to_string();
+    let daemon = Daemon::new(vec![]);
```
Comment (edouardparis) [inline tests/app_manager.rs:152]: the daemon should mock the set_spend_transaction and return the Ok response. See other tests in tests/app_managers for inspiration

**Comment by Zshan0** [comment]
As suggested, `state` is being checked with the valid `psbt`. Since the daemon is currently only dealing with a single transaction, I assumed that `setspendtx` is being called upon it and the daemon responds with an ok. Let me know if there is something that needs to be fixed. Thanks :D

File: tests/app_manager.rs [hunk 5]
```diff
@@ -145,3 +145,96 @@ async fn test_manager_create_spend() {
         .await;
     assert_eq!(sandbox.state().input_amount(), 119965368);
 }
+
+#[tokio::test]
+async fn test_manager_import_spend() {
+    let psbt_string :String = "cHNidP8BAIkCAAAAAUeuD/NEqc88sk3DoBrKoVKjXbN2xW8Jr/4GO5q87JqJAQAAAAD9////AriGJgcAAAAAIgAgSOjPZes2prPdrcgiv+IG1sjXyTCc4KDr9+C9F+xk6LwwdQAAAAAAACIAIAjkMa8elv7dHUmYpDATWBtmMmpv9yyKFawMunvGQ1AMAAAAAAABASsADicHAAAAACIAIHXyaRd0yBZ3gxhGsCgiAOKIssWXELWPdDGD1JJVB9vFAQMEAQAAAAEFR1IhAlgt7b9E9GVk5djNsGdTbWDr40zR0YAc/1G7+desKJtDIQNHBN7LVbWqiP/R710GNmJIwTFOGWVRE2/xTquLukpJDlKuIgYCWC3tv0T0ZWTl2M2wZ1NtYOvjTNHRgBz/Ubv516wom0MI1n1/6QAAAAAiBgNHBN7LVbWqiP/R710GNmJIwTFOGWVRE2/xTquLukpJDghyqV8iAAAAAAAiAgICkzqxA36tCqSnhYxtSdZwXh+zvF9msAkYr3ufAOzVJgglHWAJAAAAACICAlgt7b9E9GVk5djNsGdTbWDr40zR0YAc/1G7+desKJtDCNZ9f+kAAAAAIgIDRwTey1W1qoj/0e9dBjZiSMExThllURNv8U6ri7pKSQ4IcqlfIgAAAAAAIgICUHL04HZXilyJ1B118e1Smr+S8c1qtja46Le7DzMCaUMI+93szQAAAAAA".to_string();
+    let _psbt =
+        UnvaultTransaction::from_raw_psbt(&base64::decode(psbt_string.as_str()).unwrap()).unwrap();
```
Comment (edouardparis) [inline tests/app_manager.rs:153]: We do not use UnvaultTransaction but SpendTransaction

File: tests/app_manager.rs [hunk 6]
```diff
@@ -145,3 +145,96 @@ async fn test_manager_create_spend() {
...
+
+    let daemon = Daemon::new(vec![
+        (
+            Some(json!({"method": "listvaults", "params": Some(&[[
```
Comment (edouardparis) [inline tests/app_manager.rs:157]: we do not need the response of this request which is never triggered

File: tests/app_manager.rs [hunk 7]
```diff
@@ -145,3 +145,96 @@ async fn test_manager_create_spend() {
...
+    );
+    let sandbox = sandbox
+        .update(
+            &ctx,
+            Message::SpendTx(SpendTxMessage::PsbtEdited(psbt_string)),
+        )
+        .await;
+
+    let sandbox = sandbox
+        .update(&ctx, Message::SpendTx(SpendTxMessage::Updated(Ok(()))))
```
Comment (edouardparis) [inline tests/app_manager.rs:235]: You should first trigger the import, this message will be the result of the command. 
Do `SpendTxMessage::Import` instead

**Comment by Zshan0** [comment]
>  This request set_spend_tx must be mocked with a response.
Since `SpendTxMessage::Import` triggers `update_spend_tx` and not `set_spend_tx`, should I mock the response for this instead?

**Comment by edouardparis** [comment]
Yes your code is correct, it is 'updatespendtx' that need to be mocked.

**Pull request merged**